docs/versionhistory.rst

```diff
@@ -3,6 +3,12 @@ Version history
 
 This library adheres to `Semantic Versioning 2.0 <http://semver.org/>`_.
 
+** UNRELEASED **
+
+- Fixed a misleading ``ValueError`` in the context of DNS failures.
+  (`#580 <https://github.com/agronholm/anyio/discussions/580>`_)
+  (PR by @graingert)
```

Comment:
If this isn't an issue, you shouldn't add a link. If it is, you can convert the discussion into an issue.
```suggestion
- Fixed a misleading ``ValueError`` in the context of DNS failures (PR by @graingert)
```